metrics/network/remote_network/remote-networking-iperf3.sh

```diff
@@ -100,81 +115,103 @@ function remote_network_parallel_iperf3 {
 	start_server
 
 	client_id=$($DOCKER_EXE ps -q)
-	client_command="mount -t ramfs -o size=20M ramfs /tmp && iperf3 -c $server_ip_address -P4 -t $server_time"
-	result=$(start_client "$client_id" "$client_command")
-	total_parallel=$(echo "$result" | tail -n 4 | head -1 | awk '{print $(NF-3), $(NF-2)}')
-	echo "Parallel network is : $total_parallel"
-
+	command="iperf3 -c $server_ip_address -P4 -t $server_time"
+	check_iperf3_client_command "$command"
+	result=$(start_client "$extra_args" "$client_id" "$client_command")
+	total_parallel=$(echo "$result" | tail -n 4 | head -1 | awk '{print $(NF-3)}')
+	units=$(echo "$result" | tail -n 4 | head -1 | awk '{print $(NF-2)}')
+	echo "Parallel network is : $total_parallel $units"
+
+	save_results "$test_name" "Remote parallel" "$total_parallel" "$units"
 	clean_environment
 }
 
 # This function will measure latency across containers
 function remote_network_latency {
+	test_name="Remote network latency"
+	get_runtime
 	#Number of packages
 	number_of_packages=10
+	units="ms"
 	setup_swarm
 	client_replica_status
 	server_replica_status
 
 	server_ip_address=$(check_server_address)
 	client_id=$($DOCKER_EXE ps -q)
 	client_command="ping -c "$number_of_packages" "$server_ip_address""
-	result=$(start_client "$client_id" "$client_command")
+	result=$(start_client "$extra_args" "$client_id" "$client_command")
 	total_latency=$(echo $result | grep avg | awk '{print $(NF-1)}' | cut -d '/' -f 2)
-	echo "Network latency is : $total_latency ms"
+	echo "Network latency is : $total_latency $units"
 
+	save_results "$test_name" "Remote latency" "$total_latency" "$units"
 	clean_environment
 }
 
 function main {
+	[[ $# -ne 7 ]]&& help && die "Illegal number of parameters."
```

Comment:
I still think it would be better to check the individual items have been set after the getopts has finished. With this arg counting method you could still pass if you passed the same arg twice for instance etc.

Reply:
I agree with this point, it is better to validate individually arguments e.g. if it is mandatory to have an IP address as an argument you could do something like this:
```
if [ -z "IP_ADDRESS" ]; then
         die "no input ip addres: mandatory argument"
fi
```

Reply:
@MarioCarrilloA it has that already